0-preinstall.sh

```diff
@@ -53,10 +53,14 @@ sgdisk -Z ${DISK} # zap all on disk
 sgdisk -a 2048 -o ${DISK} # new gpt disk 2048 alignment
 
 # create partitions
-sgdisk -n 1::+1M --typecode=1:ef02 --change-name=1:'BIOSBOOT' ${DISK} # partition 1 (BIOS Boot Partition)
-sgdisk -n 2::+300M --typecode=2:ef00 --change-name=2:'EFIBOOT' ${DISK} # partition 2 (UEFI Boot Partition)
-sgdisk -n 3::-0 --typecode=3:8300 --change-name=3:'ROOT' ${DISK} # partition 3 (Root), default start, remaining
-if [[ ! -d "/sys/firmware/efi" ]]; then # Checking for bios system
+if [ -d /sys/firmware/efi ]; then
```

Comment:
! should be thrown back on if youre going to increase bios boot part and shrink efi boot part. If condition will evaluate to true for EFI.

Reply:
Where that line is the line under

> ! should be thrown back on if youre going to increase bios boot part and shrink efi boot part. If condition will evaluate to true for EFI.

The code that will run when line 56 is true is intended for the UEFI mode, not BIOS mode. I took the increased BIOS boot and shrunken efi boot from [another script](https://gist.github.com/mattiaslundberg/8620837), which also had the larger BIOS and shrunken EFI partition. And that seems to the only thing online I could find that actually fixed the booting issues in my own script. The original partition creation will be ran for legacy bios only as that has been proven to work consistently in my installer script.

Reply:
The EFIBOOT partition is mounted as /boot in the EFI install and would need the larger size to support the larger file sizes of initramfs and vmlinuz files. 
![image](https://user-images.githubusercontent.com/1778670/151421480-9e0d9369-7330-4dfb-a987-a9d01701fc88.png)

BIOS is using the smaller BIOSBOOT mount. 
![image](https://user-images.githubusercontent.com/1778670/151421544-faef7c76-6dda-42c1-b8e8-50de93999003.png)